File: nodejs/00001-標準入出力-std-in-out.js
#!/usr/bin/env node

const usage = async (status) =>{
    console.log("please stdin via pipe only.");
    console.log("Usage:");
    process.exit(status);
}

const main = async (status) =>{
    if(process.argv[2]){
        usage(1);
    }
    process.exit(status);

};

main(0);

File: R/00001-標準入出力-std-in-out.R
#!/usr/bin/env Rscript

usage<-function(status){
  print("Usage:")
  quit(status=status)
}

main<-function(status){
  args<-commandArgs(trailingOnly=TRUE)[1]

  if(!is.na(args)){
    usage(1);
  }

  input<-read.csv('stdin',header=FALSE)
  #print(as.character(input[["V1"]]))
  print(strsplit(as.character(input[["V1"]])," "))
  #print(input)
  quit(status=status)
}

main(0)
